### ISSUE TITLE
Type error on i18n.ts with nodenext moduleResolution in tsconfig (next-intl 3.0.0-beta.5)

### ISSUE THREAD

**Issue by 3Peet** [amannn/next-intl#360]
### Description

Hi, I have changed `moduleResolution` from `node` to `nodenext`
```
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "NodeNext", 👈
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "allowImportingTsExtensions": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules"]
}
```

So, I found a type error on `getRequestConfig` function in `i18n.ts`

![Screenshot 2566-06-28 at 08 22 12](https://github.com/amannn/next-intl/assets/41276225/98614454-e627-4ad7-afcc-f18aab82cba4)

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://codesandbox.io/p/sandbox/next-intl-bug-template-app-forked-g35yd6?file=%2Fi18n.ts%3A6%2C1

### Reproduction description

Steps to reproduce:
1. Change `moduleResolution` from `node` to `nodenext` in tsconfig
2. Open the `i18n.ts`
```
import { getRequestConfig } from 'next-intl/server'

export default getRequestConfig(async ({ locale }) => ({
  messages: (await import(`./messages/${locale}.json`)).default,
}))
```
3. See error: `This expression is not callable. Type 'void' has no call signatures.ts(2349)`

---
```
  "dependencies": {
    ...
    "next": "13.4.7",
    "next-intl": "3.0.0-beta.5",
    "react": "18.2.0",
    "typescript": "5.1.3",
    ...
  },
```

### Expected behaviour

The `getRequestConfig` should return correct type, not return void

**Comment by amannn** [comment]
Thanks for the report, this is correct!

This bug was already reported in https://github.com/amannn/next-intl/issues/347 and I'm planning to fix it in https://github.com/amannn/next-intl/issues/342.

I'll close this for now in favor of #342.

**Comment by 3Peet** [comment]
@amannn Thankyou!

**Comment by amannn** [comment]
I probably have a fix for this in https://github.com/amannn/next-intl/pull/371 that can be released independently of #342! 

I've reopened #347 to track this.

**Comment by snake-py** [comment]
@amannn 

Not sure if this is related but I am setting:

```json
    "moduleResolution": "Bundler",
```

and suddenly I get 
![image](https://github.com/user-attachments/assets/f18263f5-681d-4de6-92ed-326905c723df)
